aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java

```diff
@@ -0,0 +1,133 @@
+package com.nedap.archie.rminfo;
+
+import com.nedap.archie.aom.Archetype;
+import com.nedap.archie.aom.CPrimitiveObject;
+import com.nedap.archie.aom.profile.AomProfile;
+import com.nedap.archie.aom.profile.AomProfiles;
+import com.nedap.archie.base.MultiplicityInterval;
+import com.nedap.archie.rminfo.ModelInfoLookup;
+import com.nedap.archie.rminfo.RMAttributeInfo;
+import com.nedap.archie.rminfo.ReferenceModels;
+import org.openehr.bmm.core.BmmClass;
+import org.openehr.bmm.core.BmmContainerProperty;
+import org.openehr.bmm.core.BmmModel;
+import org.openehr.bmm.core.BmmProperty;
+import org.openehr.bmm.core.BmmType;
+import org.openehr.bmm.persistence.validation.BmmDefinitions;
+import org.openehr.bmm.rmaccess.ReferenceModelAccess;
+
+import java.util.List;
+
+import static org.openehr.bmm.persistence.validation.BmmDefinitions.typeNameToClassKey;
+
+/**
+ * MetaModel class that provides some opertaions for archetype validation and flattener that is either based on
+ * an implementation-derived model (ModelInfoLookup) or BMM
+ */
+public class MetaModels implements MetaModelInterface{
+
+    private final ReferenceModels models;
+    private final ReferenceModelAccess bmmModels;
+    private AomProfiles aomProfiles;
+
+    private MetaModel selectedModel;
+    private AomProfile selectedAomProfile;
+
+
+    public MetaModels(ReferenceModels models, ReferenceModelAccess bmmModels) {
+        this.models = models;
+        this.bmmModels = bmmModels;
+        aomProfiles = new AomProfiles();
+    }
+
+    public void selectModel(Archetype archetype) {
+        ModelInfoLookup selectedModel = null;
+        BmmModel selectedBmmModel = null;
+        if(models != null) {
+             selectedModel = models.getModel(archetype);
+        }
+        if(bmmModels != null) {
+             selectedBmmModel = bmmModels.getReferenceModelForClosure(BmmDefinitions.publisherQualifiedRmClosureName(archetype.getArchetypeId().getRmPublisher(), archetype.getArchetypeId().getRmPackage()));
+        }
+
+        for(AomProfile profile:aomProfiles.getProfiles()) {
+            if(profile.getProfileName().equalsIgnoreCase(archetype.getArchetypeId().getRmPublisher())) {
+                this.selectedAomProfile = profile;
+                break;
+            }
+        }
+
+        this.selectedModel = new MetaModel(selectedModel, selectedBmmModel, selectedAomProfile);
```

Comment:
So what if the model for an archetype can't be found for some reason? This will silently create a MetaModel without any model, which will return unexpected falses or throw NullPointerExceptions